python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -144,6 +146,28 @@ def _get_pre_sensor_auto_materialize_cursor(
         return deserialize_value(serialized_cursor, AssetDaemonCursor)
 
 
+def _load_asset_scoped_cursors(
+    asset_keys: AbstractSet[AssetKey], instance: DagsterInstance
+) -> Dict[AssetKey, str]:
+    """Load the cursors for a set of asset keys from the daemon cur orstorage, returning a dictionary
+    of the cursors keyed by asset key.
+    """
+    cursor_values = instance.daemon_cursor_storage.get_cursor_values(
+        {_asset_key_to_cursor_key(asset_key) for asset_key in asset_keys}
+    )
+    return {
+        _cursor_key_to_asset_key(asset_key): cursor for asset_key, cursor in cursor_values.items()
+    }
+
+
+def _asset_key_to_cursor_key(asset_key: AssetKey) -> str:
```

Comment:
I think we should have something asset-daemon-specific here. I don't think it's particularly likely that we reuse this model (storing lots of little per-asset cursors) for any other daemons, but it's better to leave ourselves the flexibility there.

Reply:
Good idea I've added a prefix